src/NServiceBus.Testing/Saga.cs

```diff
@@ -304,20 +305,36 @@ public Saga<T> When<TMessage>(Func<T, Func<TMessage, IMessageHandlerContext, Tas
         /// Invokes the saga timeout passing in the last timeout state it sent
         /// and then clears out all previous expectations.
         /// </summary>
-        /// <param name="after">The ammount of time that has passed before the timeout is called.</param>
-        public Saga<T> WhenSagaTimesOut(TimeSpan? after = null)
+        /// <param name="after">The amount of time that has passed before the timeout is called.</param>
+        public Saga<T> WhenSagaTimesOut(TimeSpan after)
         {
-            testContext.TimeoutMessages
-                .Where(t => t.Within <= after.GetValueOrDefault(TimeSpan.MaxValue))
-                .OrderBy(t => t.Within)
-                .ToList()
-                .ForEach(t => timeoutMethodInfo.Value.MakeGenericMethod(t.Message.GetType()).Invoke(this, new[]
-                {
-                    t.Message
-                }));
+            InvokeTimeouts(testContext.TimeoutMessages
+                .Where(t => t.Within.HasValue)
+                .Where(t => t.Within <= after));
 
-            testContext.Validate();
-            testContext = new TestingContext(messageCreator);
+            return this;
+        }
+
+        /// <summary>
+        /// Invokes the saga timeout passing in the date and time it sent
```

Comment:
should we state that this only invokes timeouts which have been requested at a specific DateTime (same for the one with TimeSpans)?